### DISCUSSION TITLE
Integrating an open source vector database

### DISCUSSION THREAD

**Discussion by jpchauny** [brianpetro/obsidian-smart-connections#197] [Ideas]
Hi Brian,

How about integrating an open source vector database to store the embeddings instead of a json file?

Do you think that the performance would increase significantly or is this irrelevant because of the size of an average obsidian vault? 

In order to check for AI hallucinations it would be very important to incorporate source knowledge reference metadata and this would be easily implemented by having the vector database. This could also resolve querying for a note or for a folder or for any data view metadata.

Thank you very much for the work you are doing!

**Comment by brianpetro** [comment]
> Do you think that the performance would increase significantly or is this irrelevant because of the size of an average obsidian vault?

You got it. Performance would not be significantly improved. Save the Vector DBs for enterprise scale. 

Thanks for your interest in Smart Connections,
Brian 🌴
